testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeKerberosKudu.java

```diff
@@ -0,0 +1,149 @@
+import java.util.List;
+
+import static io.trino.tests.product.launcher.docker.ContainerUtil.forSelectedPorts;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.isPrestoContainer;
+import static io.trino.tests.product.launcher.env.common.Kerberos.KERBEROS;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_PRESTO_ETC;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.BindMode.READ_ONLY;
+import static org.testcontainers.containers.BindMode.READ_WRITE;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class EnvMultinodeKerberosKudu
+        extends EnvironmentProvider
+{
+    private static final String KUDU_IMAGE = "apache/kudu:1.15.0";
+    private static final Integer KUDU_MASTER_PORT = 7051;
+    private static final Integer NUMBER_OF_REPLICA = 3;
+    private static final String KUDU_MASTER = "kudu-master";
+    private static final String KUDU_TABLET_TEMPLATE = "kudu-tserver-%s";
+
+    private static Integer initialKuduTserverPort = 7060;
+
+    private final PortBinder portBinder;
+    private final DockerFiles.ResourceProvider configDir;
+
+    @Inject
+    public EnvMultinodeKerberosKudu(PortBinder portBinder, DockerFiles dockerFiles, StandardMultinode standardMultinode, Kerberos kerberos)
+    {
+        super(standardMultinode, kerberos);
+        this.portBinder = requireNonNull(portBinder, "portBinder is null");
+        configDir = requireNonNull(dockerFiles, "dockerFiles is null").getDockerFilesHostDirectory("conf/environment/multinode-kerberos-kudu/");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        Path kerberosCredentialsDirectory = DockerFiles.createTemporaryDirectoryForDocker();
+
+        builder.configureContainer(KERBEROS, container -> container
+                .withFileSystemBind(kerberosCredentialsDirectory.toString(), "/kerberos", READ_WRITE)
+                .withCopyFileToContainer(forHostPath(configDir.getPath("kerberos_init.sh"), 0777), "/docker/kerberos-init.d/kerberos_init.sh"));
+
+        DockerContainer kuduMaster = createKuduMaster(kerberosCredentialsDirectory);
+        List<DockerContainer> kuduTablets = createKuduTablets(kerberosCredentialsDirectory, kuduMaster);
+
+        builder.addContainer(kuduMaster);
+        builder.containerDependsOn(kuduMaster.getLogicalName(), KERBEROS);
+
+        kuduTablets.forEach(tablet -> {
+            builder.addContainer(tablet);
+            builder.containerDependsOn(tablet.getLogicalName(), KERBEROS);
+        });
+
+        builder.configureContainers(container -> {
+            if (isPrestoContainer(container.getLogicalName())) {
+                configurePrestoContainer(container, kerberosCredentialsDirectory);
+                addKrb5(container);
+            }
+        });
+    }
+
+    @SuppressWarnings("resource")
```

Comment:
why are these supressions needed? I see most of the PT env definitions have these but removing them doesn't cause any warnings to appear in IDE. Are they for something like errorprone?

Reply:
I believe IntelliJ Ultimate is able to flag any "closeables" that aren't closed. Most of this annotation is used on methods that return `DockerContainer` which is a closeable.